src/lexer/mod.rs

```diff
@@ -230,7 +230,12 @@ impl Token {
                     let token_type = if is_numeric(&ident_or_number) {
                         TokenKind::BigInt(ident_or_number)
                     } else if is_hexadecimal(&ident_or_number) {
-                        TokenKind::Hex(ident_or_number)
+                        let hex = ident_or_number.trim_start_matches("0x");
+                        TokenKind::BigInt(
+                            BigInt::from_str_radix(hex, 16)
+                                .expect("invalid hex number")
+                                .to_str_radix(10),
```

Comment:
what's weird is that we're doing some parsing here, but not on line 231. WDYT? Should we instead have TokenKind::Bigint enclose a BigInt type?

Reply:
Yeah I think so, I can implement that

Reply:
I've renamed also BigInt token/expression to BigUInt, since it gets signed with `-`